### DISCUSSION TITLE
Add `Timer.is_running()`

### DISCUSSION THREAD

**Discussion by timothyqiu** [godotengine/godot-proposals#7419] [Engine Core]
To test is a `Timer` is running, the current solutions are:

* `not timer.is_stopped()` double negative twists mind
* `timer.time_left > 0`

**Comment by AThousandShips** [comment]
Would `is_running` mean `not is_stopped()` or `not is_stopped() and not is_paused()`, could easily be confusing

**Reply by Mickeon** [reply]
The latter seems like the obvious choice. `timer.time_left > 0` works just fine, even when the timer is paused.

**Reply by AThousandShips** [reply]
I'm not sure where I stand, but the former seems to be closer in line with the proposal, at minimum it'd need to be well documented, but I forsee something like this causing issues with users who don't read the documentation clearly.

And unfortunately the number of people who don't go to the documentation when something doesn't work rather than opening a bug report is not zero, so there's a risk making a method that can be hard to judge at face value

**Reply by hsandt** [reply]
I was thinking of `is_started()` = `not is_stopped()` = has started and not stopped yet (may be paused or not) for everyday use.

Then `is_running` could be `not is_stopped() and not is_paused()` = `is_started() and not is_paused()` for more specific usage.

New users would use autocompletion to discover methods, and so all the `is_` methods would be alphabetically together, and user would either compare to names to deduce the differences between `is_started` and `is_running`, or check the documentation because in this case they have no choice but to choose one of them so they need to know the difference.

**Reply by theraot** [reply]
I think adding a way to check `not is_stopped() and not is_paused()` is more useful, as it would make that check shorter and reduce roundtrips. It does not have to be `is_running`, but I can't think of a better name.

On it being confusing, I agree. And I'm reminded of Segal's Law:

> A man with a watch knows what time it is. A man with two watches is never sure.

Which in this case would mean that with `is_running` people might assume it means whatever they are thinking it means, and end up with bugs. But with `is_running` and `is_started` as @hsandt suggests, if people can't figure out which one is which, they might be confused enough to read the documentation and avoid the bug.
